zipline/assets/_assets.pyx

```diff
@@ -86,14 +88,20 @@ cdef class Asset:
         if isinstance(x, Asset):
             x_as_int = x.sid
         elif isinstance(x, int):
+            # ints are Integral, but much faster to special case
             x_as_int = x
+        elif isinstance(x, Integral):
+            x_as_int = int(x)
         else:
             return NotImplemented
 
         if isinstance(y, Asset):
             y_as_int = y.sid
         elif isinstance(y, int):
+            # ints are Integral, but much faster to special case
             y_as_int = y
+        elif isinstance(y, Integral):
```

Comment:
as we discussed, maybe make this `isinstance(y, (np.int64, np.int32, Integral))` for performance.